src/common/types/value.cpp

```diff
@@ -536,11 +559,13 @@ bool Value::TryCastAs(SQLType source_type, SQLType target_type) {
 	return true;
 }
 
-Value Value::CastAs(TypeId target_type) const {
-	if (target_type == type) {
-		return Copy(); // in case of types that have no SQLType equivalent such as POINTER
+bool Value::TryCastStrictlyAs(SQLType source_type, SQLType target_type) {
```

Comment:
Same here, please use a parameter rather than have a new function

Reply:
Thank you very much for the thorough review 👍 This all sounds very reasonable. I will have a look at the mentioned issues shortly and make some final adaptions.

Reply:
I had to change back to the CastAs method, as the TryCastAs method will change the Value itself, which was not desired in my case.